### ISSUE TITLE
Error when not creating scene 0.108.0b0

### ISSUE THREAD

**Issue by SeanPM5** [home-assistant/frontend#5403]
## Checklist

- [x] I have updated to the latest available Home Assistant version.
- [x] I have cleared the cache of my browser.
- [x] I have tried a different browser to see if it is related to my browser.

## The problem

In 0.108.0b if you try to create a Scene using the UI scene editor (by pressing the mini FAB), and then back out, it will show a toast error in the bottom:

![image](https://i.imgur.com/55t1azj.png)

Tested on another instance (0.107.7) and it does not happen there. so this is new.

Also this does not appear happen when creating a scene from Configuration -> Scenes and then backing out. It's specific to when starting from a Device detail page.

## Expected behavior

No error

## Steps to reproduce

1. Click on a device
2. Click the mini FAB on the Scenes panel to add one
3. Press the back button in top app bar to exit out, and then click YES on the "are you sure you want to leave" dialog
4. Look at bottom left for an error toast

## Environment

- Home Assistant release with the issue: 0.108.0b0
- Last working Home Assistant release (if known): 0.107.7
- Browser and browser version: Chrome
- Operating system: macOS

## Problem-relevant configuration

```yaml

```

## Javascript errors shown in your browser console/inspector

```txt

```

## Additional information

edit: this shows up in logs. Might be some kind of error in the Apple TV custom integration? If so please close this and I will report it on that repo instead.

```yaml
Log Details (ERROR)
Logger: homeassistant.components.websocket_api.http.connection.1653290256
Source: custom_components/apple_tv/media_player.py:242
Integration: websocket_api (documentation, issues)
First occurred: April 1, 2020, 8:05:36 PM (3 occurrences)
Last logged: April 1, 2020, 8:08:05 PM

Command 4 failed: 1, 0
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 134, in handle_call_service
    connection.context(msg),
  File "/usr/src/homeassistant/homeassistant/core.py", line 1232, in async_call
    await asyncio.shield(self._execute_service(handler, service_call))
  File "/usr/src/homeassistant/homeassistant/core.py", line 1255, in _execute_service
    await handler.func(service_call)
  File "/usr/src/homeassistant/homeassistant/components/homeassistant/scene.py", line 193, in apply_service
    hass, call.data[CONF_ENTITIES].values(), blocking=True, context=call.context
  File "/usr/src/homeassistant/homeassistant/helpers/state.py", line 106, in async_reproduce_state
    *(worker(domain, data) for domain, data in to_call.items())
  File "/usr/src/homeassistant/homeassistant/helpers/state.py", line 100, in worker
    hass, states_by_domain, context=context
  File "/usr/src/homeassistant/homeassistant/components/media_player/reproduce_state.py", line 98, in async_reproduce_states
    *(_async_reproduce_states(hass, state, context) for state in states)
  File "/usr/src/homeassistant/homeassistant/components/media_player/reproduce_state.py", line 65, in _async_reproduce_states
    await call_service(SERVICE_MEDIA_STOP, [])
  File "/usr/src/homeassistant/homeassistant/components/media_player/reproduce_state.py", line 55, in call_service
    DOMAIN, service, data, blocking=True, context=context
  File "/usr/src/homeassistant/homeassistant/core.py", line 1232, in async_call
    await asyncio.shield(self._execute_service(handler, service_call))
  File "/usr/src/homeassistant/homeassistant/core.py", line 1255, in _execute_service
    await handler.func(service_call)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 213, in handle_service
    self._platforms.values(), func, call, required_features
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 412, in entity_service_call
    future.result()  # pop exception if have
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 600, in async_request_call
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 443, in _handle_entity_call
    await result
  File "/config/custom_components/apple_tv/media_player.py", line 242, in async_media_stop
    await self.atv.remote_control.stop()
  File "/usr/local/lib/python3.7/site-packages/pyatv/mrp/__init__.py", line 183, in stop
    await self._send_command(CommandInfo_pb2.Stop)
  File "/usr/local/lib/python3.7/site-packages/pyatv/mrp/__init__.py", line 149, in _send_command
    command, inner.errorCode, inner.handlerReturnStatus
pyatv.exceptions.CommandError: Command 4 failed: 1, 0
```

**Comment by bramkragten** [comment]
Not able to replicate, is this from the device info page or the scenes dashboard or both?

**Comment by SeanPM5** [comment]
Seems to happen in both places after some more testing. Here is a screen recording showing it: https://streamable.com/h8zwcr

You can see in there that I didn't edit the scene at all, I just backed out, and it threw an error about applying scene or something. I have been able to reproduce it on other things too, Plex and Mobile App integration (iPhone 11) for example.

Seems like it happens on the Scenes dashboard as well (if you simply add a device then back out). But doesn't occur as often seemingly.

**Comment by bramkragten** [comment]
Is it with a specific device or with all? And does your console give you any useful info?

**Comment by SeanPM5** [comment]
This is what shows up in console, dunno if this is useful:

![console](https://i.imgur.com/qxq4G0v.png)

It does not seem to occur with _all_ devices, but quite a few to where I think it's safe to assume it's not an integration-specific issue?

And this also showed up in Dev Tools -> Logs when reproducing this with Plex.

```
Logger: homeassistant.components.websocket_api.http.connection.1680794576
Source: components/media_player/__init__.py:546
Integration: websocket_api (documentation, issues)
First occurred: 2:51:42 PM (1 occurrences)
Last logged: 2:51:42 PM

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 134, in handle_call_service
    connection.context(msg),
  File "/usr/src/homeassistant/homeassistant/core.py", line 1232, in async_call
    await asyncio.shield(self._execute_service(handler, service_call))
  File "/usr/src/homeassistant/homeassistant/core.py", line 1255, in _execute_service
    await handler.func(service_call)
  File "/usr/src/homeassistant/homeassistant/components/homeassistant/scene.py", line 193, in apply_service
    hass, call.data[CONF_ENTITIES].values(), blocking=True, context=call.context
  File "/usr/src/homeassistant/homeassistant/helpers/state.py", line 106, in async_reproduce_state
    *(worker(domain, data) for domain, data in to_call.items())
  File "/usr/src/homeassistant/homeassistant/helpers/state.py", line 100, in worker
    hass, states_by_domain, context=context
  File "/usr/src/homeassistant/homeassistant/components/media_player/reproduce_state.py", line 98, in async_reproduce_states
    *(_async_reproduce_states(hass, state, context) for state in states)
  File "/usr/src/homeassistant/homeassistant/components/media_player/reproduce_state.py", line 63, in _async_reproduce_states
    await call_service(SERVICE_MEDIA_PLAY, [])
  File "/usr/src/homeassistant/homeassistant/components/media_player/reproduce_state.py", line 55, in call_service
    DOMAIN, service, data, blocking=True, context=context
  File "/usr/src/homeassistant/homeassistant/core.py", line 1232, in async_call
    await asyncio.shield(self._execute_service(handler, service_call))
  File "/usr/src/homeassistant/homeassistant/core.py", line 1255, in _execute_service
    await handler.func(service_call)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 213, in handle_service
    self._platforms.values(), func, call, required_features
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 412, in entity_service_call
    future.result()  # pop exception if have
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 600, in async_request_call
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 443, in _handle_entity_call
    await result
  File "/usr/src/homeassistant/homeassistant/components/media_player/__init__.py", line 550, in async_media_play
    await self.hass.async_add_job(self.media_play)
  File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/src/homeassistant/homeassistant/components/media_player/__init__.py", line 546, in media_play
    raise NotImplementedError()
NotImplementedError
```

**Comment by bramkragten** [comment]
Does it only happen on `media_players`?

So it looks like an integration-specific issue, `NotImplementedError` is called when the frontend asks the backend to reset the entities to the previous state...

Nothin changed in this logic on the frontend, will check if it did in the backend.

**Comment by SeanPM5** [comment]
Not just limited to media players, it happens on the Mobile App integration too, am able to reproduce it every time: https://streamable.com/oh8o3p

I see this in logs after the mobile app errors.

```
Log Details (WARNING)
Logger: homeassistant.helpers.state
Source: helpers/state.py:96
First occurred: 1:50:50 PM (16 occurrences)
Last logged: 6:20:38 PM

Integration device_tracker does not support reproduce state
Integration sensor does not support reproduce state
```

But it also pops up the media player error at the same exact time stamp (6:20:38 PM), even though mobile_app doesn't have any media players? Very weird imo.

```
Log Details (ERROR)
Logger: homeassistant.components.websocket_api.http.connection.1695680656
Source: custom_components/apple_tv/media_player.py:242
Integration: websocket_api (documentation, issues)
First occurred: 6:20:22 PM (3 occurrences)
Last logged: 6:20:38 PM

Command 4 failed: 1, 0
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 134, in handle_call_service
    connection.context(msg),
  File "/usr/src/homeassistant/homeassistant/core.py", line 1232, in async_call
    await asyncio.shield(self._execute_service(handler, service_call))
  File "/usr/src/homeassistant/homeassistant/core.py", line 1255, in _execute_service
    await handler.func(service_call)
  File "/usr/src/homeassistant/homeassistant/components/homeassistant/scene.py", line 193, in apply_service
    hass, call.data[CONF_ENTITIES].values(), blocking=True, context=call.context
  File "/usr/src/homeassistant/homeassistant/helpers/state.py", line 106, in async_reproduce_state
    *(worker(domain, data) for domain, data in to_call.items())
  File "/usr/src/homeassistant/homeassistant/helpers/state.py", line 100, in worker
    hass, states_by_domain, context=context
  File "/usr/src/homeassistant/homeassistant/components/media_player/reproduce_state.py", line 98, in async_reproduce_states
    *(_async_reproduce_states(hass, state, context) for state in states)
  File "/usr/src/homeassistant/homeassistant/components/media_player/reproduce_state.py", line 65, in _async_reproduce_states
    await call_service(SERVICE_MEDIA_STOP, [])
  File "/usr/src/homeassistant/homeassistant/components/media_player/reproduce_state.py", line 55, in call_service
    DOMAIN, service, data, blocking=True, context=context
  File "/usr/src/homeassistant/homeassistant/core.py", line 1232, in async_call
    await asyncio.shield(self._execute_service(handler, service_call))
  File "/usr/src/homeassistant/homeassistant/core.py", line 1255, in _execute_service
    await handler.func(service_call)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 213, in handle_service
    self._platforms.values(), func, call, required_features
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 412, in entity_service_call
    future.result()  # pop exception if have
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 600, in async_request_call
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 443, in _handle_entity_call
    await result
  File "/config/custom_components/apple_tv/media_player.py", line 242, in async_media_stop
    await self.atv.remote_control.stop()
  File "/usr/local/lib/python3.7/site-packages/pyatv/mrp/__init__.py", line 183, in stop
    await self._send_command(CommandInfo_pb2.Stop)
  File "/usr/local/lib/python3.7/site-packages/pyatv/mrp/__init__.py", line 149, in _send_command
    command, inner.errorCode, inner.handlerReturnStatus
pyatv.exceptions.CommandError: Command 4 failed: 1, 0
```

So not sure whether this is a frontend issue or maybe the Apple TV beta custom integration is interfering with this somehow? cc @postlund

**Comment by balloob** [comment]
So this looks like you are calling the `scene.apply_service` and the Apple TV integration doens't like that.

**Comment by postlund** [comment]
I'm not entirely sure what `scene.apply_service` does exactly, but by the looks of it it tries to send the "stop" command which fails. By that I mean that the Apple TV returns an error code, which are the two integers "1, 0". They are quite cryptic since I haven't reversed those constants, but I decided to do that now ([here](https://github.com/postlund/pyatv/pull/557)) and it basically means `ApplicationNotFound`.

I have a theory. Normally a `playerPath` is included in most message that indicates what is playing where and by whom. Normally I don't include that, which always has defaulted to the "active player" in the past, but maybe it's time to start including that field if the device is starting to care more about it. I'm gonna try to reproduce it myself.

**Comment by postlund** [comment]
Ok, so I actually managed to reproduce this by accident. Turns out it's because of a misbehaving app. I tried "TV4 Play" and I believe they use their own media player, which doesn't plugin in to the MediaRemote stack. So no metadata is passed nor that an app is running at all to the protocol stack. So basically a "stop" command is sent to the device while nothing is playing, yield in an `ApplicationNotFound` error, which seems reasonable enough.
